erigon-lib/kv/mdbx/kv_mdbx.go

```diff
@@ -751,16 +751,16 @@ func (db *MdbxKV) BeginRo(ctx context.Context) (txn kv.Tx, err error) {
 		// otherwise carry on
 	}
 
-	if !db.trackTxBegin() {
-		return nil, fmt.Errorf("db closed")
-	}
-
 	// will return nil err if context is cancelled (may appear to acquire the semaphore)
 	if semErr := db.readTxLimiter.Acquire(ctx, 1); semErr != nil {
 		db.trackTxEnd()
```

Comment:
I think this will cause trouble

Reply:
I see what you mean now :) I was looking at the wrong lines!  Sorted this now and pushed.